configure.ac

```diff
@@ -2545,77 +2545,90 @@ fi
 # }}}
 
 # --with-libdpdk {{{
+with_libdpdk_cppflags=""
+with_libdpdk_ldflags=""
 AC_ARG_WITH(libdpdk, [AS_HELP_STRING([--with-libdpdk@<:@=PREFIX@:>@], [Path to the DPDK build directory.])],
 [
 	if test "x$withval" != "xno" && test "x$withval" != "xyes"
 	then
-		RTE_BUILD="$withval"
+		with_libdpdk_cppflags="-I$withval/include"
+		with_libdpdk_ldflags="-L$withval/lib"
 		with_libdpdk="yes"
 	else
-		RTE_BUILD="/usr"
-		with_libdpdk="$withval"
+		if test "x$withval" = "xno"
+		then
+			with_libdpdk="no (disabled)"
+		fi
 	fi
-	DPDK_INCLUDE="$RTE_BUILD/include"
-	DPDK_LIB_DIR="$RTE_BUILD/lib"
-	FOUND_DPDK=yes
 ], [with_libdpdk="no"])
 
 if test "x$with_libdpdk" = "xyes"
 then
-	LOCAL_DPDK_INSTALL="no"
-	AC_CHECK_HEADER([$DPDK_INCLUDE/rte_config.h], [LOCAL_DPDK_INSTALL=yes],
-		[AC_CHECK_HEADER([$DPDK_INCLUDE/dpdk/rte_config.h],
-		[],
-		[FOUND_DPDK=no], [])], [])
-
-	if test "x$LOCAL_DPDK_INSTALL" = "xno"
-	then
-		DPDK_INCLUDE=$DPDK_INCLUDE/dpdk
-	fi
+	SAVE_CPPFLAGS="$CPPFLAGS"
+	CPPFLAGS="$CPPFLAGS $with_libdpdk_cppflags"
+	AC_CHECK_HEADERS(rte_config.h, [with_libdpdk="yes"],
+		[
+			AC_CHECK_HEADERS(dpdk/rte_config.h,
+				[
+					with_libdpdk="yes"
+					with_libdpdk_cppflags="$with_libdpdk_cppflags/dpdk"
```

Comment:
I don't think you can. As you said yourself, since the headers in /usr/include/dpkd depend on other headers in that same directory, you need to have /usr/include/dpkd in your include path. And how about on other operating systems, for instance on FreeBSD where includes are in /usr/local/include?